services/visual-studio-marketplace/visual-studio-marketplace-downloads.service.js

```diff
@@ -42,27 +42,53 @@ module.exports = class VisualStudioMarketplaceDownloads extends VisualStudioMark
         namedParams: { extensionId: 'ritwickdey.LiveServer' },
         staticPreview: this.render({ measure: 'd', count: 1239 }),
         keywords: this.keywords,
+        documentation: `For Azure DevOps extensions, this includes total downloads for both on-prem Azure DevOps Server and Azure DevOps Services`,
```

Comment:
As someone not familiar with these tools, the second half of this isn't super clear. What does it mean "for both on-prep Azure DevOps Server and Azure DevOps Services?"

Reply:
As far as Azure DevOps Server & Azure DevOps Services in this context it is the same concept as Bitbucket Server & Bitbucket Cloud, GitHub Enterprise & GitHub, Shields.io vs. self-hosted Shields 😄 etc. Services refers to the SaaS/cloud offering, Server is the one you download, install, and run yourself.

It's a concept that will definitely be familiar to publishers and consumers: it's the labeling used on extensions in the marketplace, every extension publisher has to specify whether their extension is supported for Azure DevOps Services and/or Server, and consumers acquire the extension specifically for one of those two targets. 

![image](https://user-images.githubusercontent.com/13042488/51291254-1baf3200-19cc-11e9-92fd-8ac765d37a75.png)

I'm definitely open to any suggested changes on the wording, but I do think the description is important.

Reply:
I could try to suggest an improvement. What does it mean to "download" these extensions?

Reply:
When a user wants to acquire an extension for Azure DevOps, they start from the marketplace but the way they go about actually getting that extension varies noteably between the SaaS and self-hosted platforms. 

For SaaS, acquiring an extension basically happens via a click/confirmation process by the user and the extension is installed in that users account automagically by the platform.

For self-hosted, the user "acquires" the extension by downloading the extension package from the marketplace (a `.vsix` package file if I recall correctly) to their local machine and from there they add that extension to their self-hosted Azure DevOps server.

Reply:
Honestly it may make more sense to handle the badges for extensions targeting Azure DevOps differently although that would require the user specifying their target was for Azure DevOps (which we were initially trying to avoid) 

In theory there's 3 data points for Marketplace extensions that target Azure DevOps which we could provide badges for:

1) Number of times the extension has been acquired for SaaS target
2) Number of times the extension has been acquired for Self-hosted target
3) Total of 1 + 2

Reply:
Heh. It's still kind of confusing, isn't it.

For Server, is the act of downloading the `.vsix` what gets counted?

For Services, is it the install that gets counted?

Reply:
> * Number of times the extension has been acquired for SaaS target
> * Number of times the extension has been acquired for Self-hosted target

I was going to suggest a second badge, though we do already have the Visual Studio Marketplace Installs badge. Is downloads vs installs making a similar distinction?

Reply:
> Heh. It's still kind of confusing, isn't it.

Yeah it is. Even as someone who has published extensions on that marketplace.. I still have questions about some of the metrics/labels (there's a deeper layer viewable in the publisher portal that is even more confusing). 


> For Server, is the act of downloading the `.vsix` what gets counted?
> 
> For Services, is it the install that gets counted?

Yup. The way I've been trying to look at is what info matters to an extension publisher (I presume the main target audience for badges), and match that with the data available to Shields via the marketplace API. 

The API response has two relevant numerical data points for an extension, Which I find to be a straightforward distinction:

`install` - the number of times the extension was added to SaaS instance
`onPremDownloads` - the number of times the extension was added to a self-hosted 

The part that makes it all the more confusing is that the Marketplace UI lists `installs` as the total of those two values (SaaS + Self-hosted) so that it is consistent with what it shows for VS Code and VS Extensions

Am I making this worse? 🤔